components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -979,6 +982,37 @@ public static boolean jsFunction_updateAPIDesign(Context cx, Scriptable thisObj,
         api.setAccessControl(publisherAccessControl);
         api.setAccessControlRoles(publisherAccessControlRoles);
 
+        FileHostObject wsdlFile;
+        if (apiData.containsKey(APIConstants.WSDL_FILE)) {
+            wsdlFile = (FileHostObject) apiData.get(APIConstants.WSDL_FILE, apiData);
+            if (wsdlFile != null) {
+                if (wsdlFile.getName().endsWith(APIConstants.ZIP_FILE_EXTENSION)) {
+                    WSDLArchiveInfo archiveInfo = APIUtil.extractAndValidateWSDLArchive(wsdlFile.getInputStream());
+                    if (archiveInfo != null) {
+                        api.setWsdlArchivePath(archiveInfo.getAbsoluteFilePath());
+                        if (log.isDebugEnabled()) {
+                            log.debug("WSDL Archive in the file path " + archiveInfo.getAbsoluteFilePath()
+                                    + "is extracted and validated.");
+                        }
+                        ResourceFile wsdlResource = new ResourceFile(wsdlFile.getInputStream(),
+                                wsdlFile.getJavaScriptFile().getContentType());
+                        api.setWsdlArchive(wsdlResource);
+                        APIFileUtil.deleteDirectory(archiveInfo.getLocation());
+                    }
+                } else if (wsdlFile.getName().endsWith(APIConstants.WSDL_FILE_EXTENSION)) {
+                    String path = System.getProperty(APIConstants.JAVA_IO_TMPDIR) + File.separator
+                            + APIConstants.WSDL_ARCHIVES_TEMP_FOLDER + File.separator + UUID.randomUUID().toString();
+                    String wsdlFilePath = path + File.separator + APIConstants.WSDL_FILE
+                            + APIConstants.WSDL_FILE_EXTENSION;
+                    APIFileUtil.extractSingleWSDLFile(wsdlFile.getInputStream(), path, wsdlFilePath);
+                    api.setWsdlUrl(APIConstants.FILE_URI_PREFIX + wsdlFilePath);
+                } else {
+                    throw new APIManagementException("Invalid file archive file extension: Use .zip format");
```

Comment:
Shall we change this error message as "Could not extract the archived file.  The file must be a ZIP compressed file"